ntbs-service/Services/LegacySearchService.cs

```diff
@@ -61,58 +73,37 @@ public async Task<(IEnumerable<Notification> notifications, int count)> Search(i
 
                 count = (await connection.QueryAsync<int>(CountQuery)).Single();
             }
-            return (results.Select(AsNotification), count);
+            var notificationBannerModels = results.Select(r => (NotificationBannerModel)AsNotificationBannerAsync(r).Result);
+            return (notificationBannerModels, count);
         }
 
-        private static Notification AsNotification(dynamic result)
+        private async Task<NotificationBannerModel> AsNotificationBannerAsync(dynamic result)
         {
-            var notification = new Notification
+            TBService tbService = null;
+            if(result.NtbsHospitalId is Guid guid)
+            {
+                tbService = await _referenceDataRepository.GetTbServiceFromHospitalIdAsync(guid); 

```

Comment:
do we want to call `Result` here and make this method not `async`? It would make the `Select` call above a bit less clunky, but we might need to have a think about whether it impacts anything to make that switch..

Reply:
I'm happy to swap to this - I'm not sure myself which way looks better

Reply:
Was the hope that this might mean the cast in the select statement above isn't necessary? As that didn't happen (would remove the .Result though)